src/apify_client/clients/base/base_client.py

```diff
@@ -0,0 +1,37 @@
+from typing import Any, Dict
+
+from ..._http_client import _HTTPClient
+from ..._utils import _to_safe_id
+
+
+class BaseClient:
+    """Base class for sub-clients."""
+    def __init__(self, *, base_url: str, http_client: _HTTPClient, resource_id: str = None, resource_path: str, params: Dict = None) -> None:
+        """Initializes the sub-client.
+
+        Args:
+            base_url (str): Base URL of the API server
+            http_client (_HTTPClient): The _HTTPClient instance to be used in this client
+            resource_id (str): ID of the manipulated resource, in case of a single-resource client
+            resource_path (str): Path to the resource's endpoint on the API server
+            params (dict): Parameters to include in all requests from this client
+        """
+        self.base_url = base_url
+        self.http_client = http_client
+        self.params = params or {}
+        self.resource_path = resource_path
+        self.resource_id = resource_id
+        self.url = f'{self.base_url}/{self.resource_path}'
```

Comment:
Can we enforce that `url` never ends with `/`? Otherwise it would start making problems due to double `/` (saw this before).